gr/lessons/advanced/gen-stage.md

```diff
@@ -11,39 +11,49 @@ title: GenStage
 
 Τι είναι το GenStage; Από την επίσημη τεκμηρίωση, είναι "ένας προσδιορισμός και μια επεξεργαστική ροή για την Elixir", αλλά τι σημαίνει αυτό για εμάς;
 
-Σημαίνει ότι το GenStage μας παρέχει έναν τρόπο να ορίζουμε έναν αγωγό εργασίας που εκτελείται από ανεξάρτητα βήματα (ή στάδια) σε ξεχωριστές διεργασίες.  Αν έχετε δουλέψει με αγωγούς στο παρελθόν τότε μερικές από τις έννοιες θα σας είναι γνώριμες.
+Σημαίνει ότι το GenStage μας παρέχει έναν τρόπο να ορίζουμε έναν αγωγό εργασίας που εκτελείται από ανεξάρτητα βήματα (ή στάδια) σε ξεχωριστές διεργασίες·  Αν έχετε δουλέψει με αγωγούς στο παρελθόν τότε μερικές από τις έννοιες θα σας είναι γνώριμες.
 
 Για να κατανοήσετε καλύτερα πως δουλεύει, ας σκεφτούμε μια απλή ροή παραγωγού-καταναλωτή:
 
 ```
 [A] -> [B] -> [C]
 ```
 
-Σε αυτό το παράδειγμα έχουμε τρία στάδια: το `A` είναι ένας παραγωγός, το `B` είναι ένας παραγωγός-καταναλωτής και το `C` είναι ένας καταναλωτής.  το `A` παράγει μια τιμή η οποία καταναλώνεται από το `B`, το `B` κάνει κάποια εργασία και επιστρέφει μια νέα τιμή η οποία λαμβάνεται από τον καταναλωτή μας `C`.  Ο ρόλος του σταδίου μας είναι σημαντικός, όπως θα δούμε στον επόμενο τομέα.
+Σε αυτό το παράδειγμα έχουμε τρία στάδια: το `A` είναι ένας παραγωγός, το `B` είναι ένας παραγωγός-καταναλωτής και το `C` είναι ένας καταναλωτής.
+Το `A` παράγει μια τιμή η οποία καταναλώνεται από το `B`, το `B` κάνει κάποια εργασία και επιστρέφει μια νέα τιμή η οποία λαμβάνεται από τον καταναλωτή μας `C`·  Ο ρόλος του σταδίου μας είναι σημαντικός, όπως θα δούμε στον επόμενο τομέα.
 
 Παρόλο που το παράδειγμά μας είναι 1-προς-1 παραγωγός-προς-καταναλωτή, είναι δυνατό να έχουμε πολλαπλούς παραγωγούς και πολλαπλούς καταναλωτές σε οποιοδήποτε στάδιο.
 
-Για να παρουσιάσουμε καλύτερα αυτές τις έννοιες θα κατασκευάσουμε έναν αγωγό με το GenStage, αλλά πρώτα ας εξερευνήσουμε περισσότερο τους ρόλους στους οποίος βασίζεται το GenStage.
+Για να παρουσιάσουμε καλύτερα αυτές τις έννοιες θα κατασκευάσουμε έναν αγωγό με το GenStage, αλλά πρώτα ας εξερευνήσουμε περισσότερο τους ρόλους στους οποίους βασίζεται το GenStage.
 
 ## Καταναλωτές και Παραγωγοί
 
-Όπως έχουμε διαβάσει, ο ρόλος που δίνουμε στο στάδιό μας είναι σημαντικός. Οι προδιαγραφές του GenStage αναγνωρίζουν τρεις ρόλους:
+Όπως έχουμε διαβάσει, ο ρόλος που δίνουμε στο στάδιό μας είναι σημαντικός.
+Οι προδιαγραφές του GenStage αναγνωρίζουν τρεις ρόλους:
 
-+ `:producer` — Μια πηγή.  Οι παραγωγοί περιμένουν για ζήτηση από τους καταναλωτές και ανταποκρίνονται με τα συμβάντα που ζητήθηκαν.
++ `:producer` — Μια πηγή.
+Οι παραγωγοί περιμένουν για ζήτηση από τους καταναλωτές και ανταποκρίνονται με τα συμβάντα που ζητήθηκαν.
```

Comment:
I have to check if this works prorperly as, it might break this list. Let me get back to you for this one.